### DISCUSSION TITLE
Feature Request: Persistent Dark Mode Setting

### DISCUSSION THREAD

**Discussion by GrizzlyAK** [zwave-js/zwave-js-ui#2041] [Ideas]
Would it be possible to move the Dark Mode option to the Settings panel so that it can be persistent. Cookies get destroyed (or not saved) and the mode keeps getting reset.

**Comment by robertsLando** [comment]
@GrizzlyAK It's already stored on localstorage, are you using incognito mode or something else that doesn't persists local storage for some reason?

**Reply by GrizzlyAK** [reply]
Yes, I use Private Browsing in FF at all times (doesn't everyone?)  This used to happen with the HA UI as well, until they changed it to use the system (or Backend-selected) theme setting instead. Maybe that is an option, or just put it on disk storage? Not a biggie, just a minor annoyance.

**Reply by robertsLando** [reply]
@GrizzlyAK Localstorage is the easier way to do this, we also store the table filters, columns and many other preferences there, couldn't you enable localstorage only for that site using some options?

**Reply by GrizzlyAK** [reply]
Hmm... interesting idea, but I'm not sure FF would do that in private browsing mode. As it is now, every time I first open the Control Panel, I have to click a button in the address bar (enhanced tracking protection), and then flip a switch in the resulting dialog just to _show_ the Control Panel. Otherwise, I get a blank page. Not sure why it won't show it, and even though I turn it off once, it forces me to do it again if I restart FF. But that's really a different issue. As long as I have it open, I can reopen the CP in a new tab and it remembers that setting as well as my dark mode and table filters, etc. But as soon as I close FF, the slate is clean and I have to start over. I will explore it further but I doubt FF allows for that level of detail in controlling persistence in private mode.

![image](https://user-images.githubusercontent.com/4411963/145957233-7de7adb4-9c1a-4686-a58d-e7e200346efa.png)

**Reply by robertsLando** [reply]
Maybe this can help: https://support.mozilla.org/en-US/questions/1280262

**Reply by GrizzlyAK** [reply]
I don't think that will help. I think the info box in the Settings shown below is the issue with this approach (for me). I don't use private browsing mode by tab, but have it turned on all the time. I've been doing that for years, and just deal with the loss of persistence. I just thought that since you were already saving things on disk per-user, that this setting could be included. Again, nbd.

![image](https://user-images.githubusercontent.com/4411963/146094704-f7a8e5f8-b08e-405c-8694-f80eb764d9e0.png)

**Reply by robertsLando** [reply]
Yeah let me open a feature request for that so :)

**Reply by robertsLando** [reply]
https://github.com/zwave-js/zwavejs2mqtt/issues/2058
